src/api/dimension/DimensionBigBlueButtonService.ts

```diff
@@ -163,6 +170,225 @@ export class DimensionBigBlueButtonService {
         return {url: joinUrl};
     }
 
+    /**
+     * Clients can call this endpoint in order to retrieve the contents of the widget room state.
+     * This endpoint will create a BigBlueButton meeting and place the returned ID and password in the room state.
+     * @param {string} roomId The ID of the room that the widget will live in.
+     */
+    @GET
+    @Path("widget_state")
+    public async widget(
+        @QueryParam("roomId") roomId: string,
+    ): Promise<BigBlueButtonWidgetResponse|ApiError> {
+        // Hash the room ID in order to generate a unique widget ID
+        const widgetId = sha256(roomId + "bigbluebutton");
+
+        const widgetName = config.bigbluebutton.widgetName;
+        const widgetTitle = config.bigbluebutton.widgetTitle;
+        const widgetAvatarUrl = config.bigbluebutton.widgetAvatarUrl;
+
+        LogService.info("BigBlueButton", "Got a meeting create request for room: " + roomId);
+
+        // NOTE: BBB meetings will by default end a minute or two after the last person leaves.
+        const createQueryParameters = {
+            meetingID: randomString(20),
+            // To help admins link meeting IDs to rooms
+            meta_MatrixRoomID: roomId,
+        };
+
+        // Create a new meeting.
+        const createResponse = await this.makeBBBApiCall("GET", "create", createQueryParameters, null);
+        LogService.info("BigBlueButton", createResponse);
+
+        // The password users will join with.
+        // TODO: We could give users access to moderate the meeting if we returned createResponse.moderatorPW, but it's
+        // unclear how we pass this to the user without also leaking it to others in the room.
+        // We could have the client request the password and depending on their power level in the room, return either
+        // the attendee or moderator one.
+        const attendeePassword = createResponse.attendeePW[0];
+
+        // TODO: How do we get the user dimension is actually running as?
```

Comment:
`MatrixStickerBot.getUserId()` - the account is a general purpose utility account, but up until nowish only had a purpose of idling in stickerpack rooms.